File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/dominio/termo/MaiorRemuneracaoProxy.java
/*
 * Decompiled with CFR 0.152.
 */
package br.jus.trt8.pjecalc.negocio.dominio.termo;

import br.jus.trt8.pjecalc.base.comum.Utils;
import br.jus.trt8.pjecalc.negocio.dominio.termo.ParametroDoTermo;
import br.jus.trt8.pjecalc.negocio.dominio.termo.Termo;
import java.math.BigDecimal;

public class MaiorRemuneracaoProxy
implements Termo {
    private static final long serialVersionUID = 6020534453701170096L;

    @Override
    public BigDecimal resolverValor(ParametroDoTermo parametro) {
        return parametro.getValorMaiorRemuneracaoDoCalculo();
    }

    public String toString() {
        return Utils.formatarNumero(this.resolverValor(null));
    }
}

File: src/lib/pjecalc/core/dominio/termo/maior-remuneracao-proxy.ts
/**
 * Porte 1:1 de MaiorRemuneracaoProxy.java (24 linhas).
 * Delega para Calculo.getValorMaiorRemuneracao.
 *
 * Ref: pjecalc-fonte/.../dominio/termo/MaiorRemuneracaoProxy.java
 */
import type Decimal from 'decimal.js';
import type { Termo } from './termo';
import type { ParametroDoTermo } from './parametro-do-termo';

export class MaiorRemuneracaoProxy implements Termo {
  resolverValor(parametro: ParametroDoTermo): Decimal {
    return parametro.getValorMaiorRemuneracaoDoCalculo();
  }
}
